app/components/ui/files/destinate-folder-modal/component.js

```diff
@@ -12,19 +13,19 @@ export default Ember.Component.extend({
   userAuth: service(),
   folderNavs: service(),
 
-  folders: Ember.A(), //Array of folders in the directory
+  folders: A(), //Array of folders in the directory
```

Comment:
Is this syntax necessary to create Arrays in Ember?

From what I understand, Ember extends raw arrays (e.g. `[]`) by default unless `EmberENV.EXTEND_PROTOTYPES` is set to false - in our case, it is not currently false. Going forward, we shouldn't need the `A([])` syntax unless that value changes.

See https://www.emberjs.com/api/ember/3.7/functions/@ember%2Farray/A

Reply:
Taken from [Ember documentation](https://guides.emberjs.com/v3.2.0/object-model):

> JavaScript objects don't support the observation of property value changes. Consequently, if an object is going to participate in Ember's binding system you may see an Ember.Object instead of a plain object.

> Ember.Object also provides a class system, supporting features like mixins and constructor methods. Some features in Ember's object model are not present in JavaScript classes or common patterns, but all are aligned as much as possible with the language and proposed additions.

> Ember also extends the JavaScript Array prototype with its Ember.Enumerable interface to provide change observation for arrays.

So, yes, I'd say that syntax is necessary if you want to observe the properties within the elements of any given array at some point. The price to pay for wrapping those arrays is pretty low when compared to the binding benefits that can be drawn from it.

Reply:
I would agree if there were additional benefits over the default. I may be misinterpreting your quoted text:
> Ember also extends the JavaScript Array prototype with its Ember.Enumerable interface to provide change observation for arrays.

That last sentence indicates that we don't need the special array, as Ember will extend JavaScript's built-in `Array.prototype` to provide the same change observation. This matches what I've said above about `EXTEND_PROTOTYPES`, which we can currently leverage to simplify our syntax and not lose those benefits. I will do some more research to confirm whether this is the case and try to come up with a simple example.

It may also be worth noting that change observation and event handling can add performance overhead if it is not necessary.